ios/Classes/SwiftFlutterWeatherSensorPlugin.swift

```diff
@@ -1,14 +1,76 @@
 import Flutter
 import UIKit
+import Combine
 
-public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin {
+@available(iOS 13.0, *)
+public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+  private var eventSink: FlutterEventSink? = nil
+  private var sinkSubscription: AnyCancellable?
+    
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let methodChannel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let eventChannel = FlutterEventChannel(name: "flutter_weather_sensor_plugin/readings", binaryMessenger: registrar.messenger())
+      
     let instance = SwiftFlutterWeatherSensorPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: methodChannel)
+    eventChannel.setStreamHandler(instance)
```

Comment:
Just curious - do any of these automatic (on-the-stack) vars need to maintain scope outside of this method? I.e., should they be instance vars?

Reply:
I am not 100% sure. I would have guessed they should be instance variables as well but the boilerplate code created when creating a new flutter plugin set it up like this, so I kept it like that. I can update it though if you think that is necessary.

Reply:
It's not a very big deal, but it would be nice to not have to rely upon the boilerplate example, just in case it is not so great. I could help you with this if you'd like. Looking at this some more, the only strange thing is eventChannel. Both methodChannel and instance are passed over to `registrar`, but of course it's not clear who or how a reference to eventChannel is being held onto. Oh well, let's just leave it.